service/resource/cloudformation/current.go

```diff
@@ -1,7 +1,54 @@
 package cloudformation
 
-import "context"
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	awscloudformation "github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/giantswarm/aws-operator/service/key"
+	"github.com/giantswarm/microerror"
+)
 
 func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
-	return nil, nil
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.Log("cluster", key.ClusterID(customObject), "debug", "looking for AWS stack")
+
+	stackName := key.MainStackName(customObject)
+
+	describeInput := &awscloudformation.DescribeStacksInput{
+		StackName: aws.String(stackName),
+	}
+	describeOutput, err := r.awsClient.DescribeStacks(describeInput)
+
+	// FIXME: The validation error returned by the CloudFormation API doesn't make things easy to check, other than
+	// looking for the returned string. There's no constant in aws go sdk for defining this string, it comes from
+	// the service.
+	stackNotFoundError := fmt.Sprintf("Stack with id %s does not exist", stackName)
+	if strings.Contains(err.Error(), stackNotFoundError) {
```

Comment:
Even though it is not nice by the client lib we use it would be good to have an error matcher like we have for our own errors on our side. It would then just be `if IsStackNotFound(err)`. I would then also only check for a substring and make sure the string `does not exist` is contained. Tomorrow they decide to change their error messages, then the matcher would fail because we compare against some too strict string.

Reply:
Done